numpy/f2py/cfuncs.py

```diff
@@ -549,7 +549,12 @@
 #define F2PY_THREAD_LOCAL_DECL __declspec(thread)
 #elif defined(__STDC_VERSION__) \\
       && (__STDC_VERSION__ >= 201112L) \\
-      && !defined(__STDC_NO_THREADS__)
+      && !defined(__STDC_NO_THREADS__) \\
+      && (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ > 12))
```

Comment:
Does this also need to handle cases where glibc is not defined at all?

Reply:
Yes, fixed.